projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-text-field/dot-binary-text-field.tsx

```diff
@@ -0,0 +1,182 @@
+import Fragment from 'stencil-fragment';
+import { DotBinaryTextStatusEvent, DotFieldStatus, DotFieldValueEvent } from '../../../models';
+import {
+    getErrorClass,
+    getOriginalStatus,
+    isFileAllowed,
+    isValidURL,
+    updateStatus
+} from '../../../utils';
+
+/**
+ * Represent a dotcms text field for the binary file element.
+ *
+ * @export
+ * @class DotBinaryFile
+ */
+@Component({
+    tag: 'dot-binary-text-field',
+    styleUrl: 'dot-binary-text-field.scss'
+})
+export class DotBinaryTextFieldComponent {
+    @Element() el: HTMLElement;
+
+    /** Value specifies the value of the <input> element */
+    @Prop({ mutable: true, reflectToAttr: true })
+    value = null;
+
+    /** (optional) Placeholder specifies a short hint that describes the expected value of the input field */
+    @Prop({ reflectToAttr: true })
+    placeholder = 'Attach files by dragging & dropping, selecting or pasting them.';
+
+    /** (optional) Determine if it is mandatory */
+    @Prop({ reflectToAttr: true })
+    required = false;
+
+    /** (optional) Array that describes a type of file that may be selected by the user, eg: .pdf,.jpg  */
+    @Prop({ reflectToAttr: true })
+    accept = '';
+
+    /** (optional) Disables field's interaction */
+    @Prop({ reflectToAttr: true })
+    disabled = false;
+
+    @State() status: DotFieldStatus;
+
+    @Event() _valueChange: EventEmitter<DotFieldValueEvent>;
+    @Event() _statusChange: EventEmitter<DotBinaryTextStatusEvent>;
+
+    private errorType = '';
+    private file = null;
+    private allowedFileTypes = [];
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     */
+    @Method()
+    reset(): void {
+        this.value = '';
+    }
+
+    @Watch('accept')
+    optionsWatch(): void {
+        this.allowedFileTypes = !!this.accept ? this.accept.split(',') : [];
+    }
+
+    componentWillLoad(): void {
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+    }
+
+    render() {
+        return (
+            <Fragment>
+                <input
+                    type="text"
+                    class={getErrorClass(this.status.dotValid)}
+                    onBlur={() => this.blurHandler()}
+                    disabled={this.disabled}
+                    placeholder={this.placeholder}
+                    value={this.value}
+                    onKeyDown={(event: KeyboardEvent) => this.keyDownHandler(event)}
+                    onKeyPress={(event: KeyboardEvent) => event.preventDefault()}
+                    onPaste={(event: ClipboardEvent) => this.pasteHandler(event)}
+                />
+            </Fragment>
+        );
+    }
+
+    // only supported in macOS.
+    private pasteHandler(event: ClipboardEvent): void {
+        this.value = '';
+        const clipboardData: DataTransfer = event.clipboardData;
+        if (clipboardData.items.length) {
+            const clipBoardFileName = clipboardData.items[0];
```

Comment:
What is `item[0]`? maybe something like:
```suggestion
            const [someName] = clipboardData.items;
```